### DISCUSSION TITLE
ランタイム間の比較方法

### DISCUSSION THREAD

**Discussion by k2-kanda** [openstandia/ng-js-runtime-docs#71] [General]
最初は
- Node.js
- Deno
- Bun

の三者を比較する方針であった。
比較項目を洗い出すのはそうするとして、各ランタイム個別の特徴があればそれも別途書く？

そこから踏み込んで、DenoやBunが向いているユースケースを進めたい。（Node.jsには戻らないとして）
ここはスクラッチであることを前提にしてもいいかもしれない。Node.jsプロジェクトからのマイグレーションだと限界があると思うので。

**Comment by jumpei-kanno** [comment]
### RAでのビルド速度の比較まとめ
以前の計測ではNext.jsのビルドキャッシュが影響してビルド時間の測定がきちんと行われていなかった可能性があるため、キャッシュを考慮して再度実行

|runtime|ビルド時間（キャッシュ無）|ビルド時間（キャッシュ有）|
|-|-|-|
|node|1m55.280|1m1.143|
|deno|1m29.887|38.867|
|deno（npm指定）|1m31.446|43.393|
|bun|1m42.505|46.551|


ちなみに、DenoキャッシュはNext.jsのビルドには影響をあまり与えなかった。
denoとbunに関しては計測結果の差異をあまり感じられなかった。（少しdenoの方が良いくらい）
